### DISCUSSION TITLE
Variable to allow all possible mount values in state parameter

### DISCUSSION THREAD

**Discussion by gavillom** [bluebanquise/bluebanquise#435] [Ideas]
Hi all,

I've discussed with @oxedions about PR #412. On my side, I'm happy to see new variable _nfs_client_directories_state_ that would help to use this nfs_client role in diskless creation step (that wasn't possible before).

On my team, we've developed such a Lustre client role that is quite similar to nfs_client role and help us to also mount Lustre filesystem. I'll implement such variable on this role. And I think it would be interesting that variable name is the same than for nfs_client role. So that I would just have to set this variable to manage mount state for both roles at a time.

What do you think about it ? 
One suggestion for variable name would be _fs_client_directories_state_.

**Comment by btravouillon** [comment]
We may lose granularity. However, it should be OK with (top of my head) `<fstype>_client_directories_state | default(fs_client_directories_state) | default('the default value we want')`. We may have to remove `nfs_client_directories_state` from `default/main.yml`.
Also, maybe `fs_client_mount_state` could be more relevant?

**Reply by gavillom** [reply]
Sounds good to me. 
I agree with _fs_client_mount_state_.

**Reply by btravouillon** [reply]
BTW, I am working on an Ansible Collection for Lustre on my spare time. The code is still private, I may upload it soon. It might be of interest to share our efforts.

**Reply by gavillom** [reply]
Sure :-)

**Comment by oxedions** [comment]
So this should be added with services stuff in inventory/group_vars/all/general_settings/general.yml

I think putting that into the inventory/group_vars/all/general_settings/nfs.yml file may make people think it is only related to nfs and nothing else (even if lustre is an network file system in a way...)

**Reply by gavillom** [reply]
Agreed.
